sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -1348,10 +1348,12 @@ std::unique_ptr<CurlNetworkConnection> CurlConnectionPool::ExtractOrCreateCurlCo
   }
 
   long sslOption = 0;
+#if LIBCURL_VERSION_MAJOR >= 7 && LIBCURL_VERSION_MINOR >= 44
```

Comment:
nit: I know we don't have libcurl 8 (or higher) yet, but wouldn't something like 8.0 result in this condition being false, incorrectly?

We may want something like this:
```suggestion
#if (LIBCURL_VERSION_MAJOR == 7 && LIBCURL_VERSION_MINOR >= 44) || LIBCURL_VERSION_MAJOR > 7
```

Reply:
if you look at [curlver.h](https://github.com/curl/curl/blob/master/include/curl/curlver.h), there's a helper macro we can leverage.

```suggestion
#if (CURL_AT_LEAST_VERSION(7,44,0))
```

Reply:
That would have been nice, but they are not available in the curl shipped with CentOS7

Reply:
I updated the check to be
`#if !(LIBCURL_VERSION_MAJOR <= 7 && LIBCURL_VERSION_MINOR < 44)`